### PR TITLE
Transition to using only tiles in BHCal

### PR DESCRIPTION
**Pull request by ruse-traveler**
### Briefly, what does this PR introduce?

This PR removes the merging of BHCal tiles during digitization, and modifies the BHCal adjacency matrix such that the algorithm clusters _tiles_ rather than _towers_.

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [X ] New feature (issue #__)
- [ ] Documentation update
- [ ] Other: __

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [X ] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?

No.

### Does this PR change default behavior?

Yes. Reconstructed hits in the context of the BHCal will now be _tiles_ (of which there are 5 per tower) rather _towers_. Furthermore, island clusters will be formed from tiles rather towers.

### CODE REVIEW

File: src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h [hunk 1]
```diff
@@ -38,9 +38,11 @@ class ProtoCluster_factory_HcalBarrelIslandProtoClusters : public JFactoryT<edm4
...
         //   24 - number of towers per row
+        //   5  - number of tiles per tower
         u_adjacencyMatrix =
           "("
-          "  abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
+          "  abs(fmod(tile_1, 5) - fmod(tile_2, 5))"
+          "  + abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
           "  + min("
           "      abs((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24)),"
           "      32 * 2 - abs((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24))"
```
Comment (veprbl) [inline src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h:48]:
```suggestion
                    "      abs(((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 + fmod(tile_1, 5) - fmod(tile_2, 5)),"
          "      32 * 2 - abs(((sector_1 - sector_2) * 2 + floor(tower_1 / 24) - floor(tower_2 / 24)) * 5 + fmod(tile_1, 5) - fmod(tile_2, 5))"
```

File: src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h [hunk 2]
```diff
@@ -38,9 +38,11 @@ class ProtoCluster_factory_HcalBarrelIslandProtoClusters : public JFactoryT<edm4
         //   32 - number of sectors
         //   2  - number of rows per sector
         //   24 - number of towers per row
+        //   5  - number of tiles per tower
         u_adjacencyMatrix =
           "("
-          "  abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
+          "  abs(fmod(tile_1, 5) - fmod(tile_2, 5))"
+          "  + abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
```
Comment (veprbl) [inline src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h:45]:
```suggestion
          "  abs(fmod(tower_1, 24) - fmod(tower_2, 24))"
```

File: src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h (hunk 1 above)
Comment (ruse-traveler) [reply src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h]: I checked this for a few cases, and a think the `sector_1 - sector_2` terms need an extra 5 in front... For example, as written the expression works out to 7 for `(tile, tower, sector) = (4, 27, 20)` vs. `(0, 3, 1)` (which should be adjacent), and it works out to 71 for `(4, 47, 31)` vs. `(0, 23, 0)` (which also should be adjacent).

Adjusting it to `2 * 5 * abs(sector_1 - sector_2)`  makes both cases to 1!

Comment (veprbl) [reply src/detectors/HCAL/ProtoCluster_factory_HcalBarrelIslandProtoClusters.h]: Yes, there were brackets: `(x * 2 + y) * 5 + z`

**Review by veprbl — APPROVED** [review]
LGTM

**Pull request merged**